3DAmsterdam/Assets/Amsterdam3D/Scripts/Interface/ContextPointerMenu/ContextPointerMenu.cs

```diff
@@ -1,104 +1,127 @@
+			foreach (Button button in buttons)
+				button.onClick.AddListener(CloseContextMenu);
+		}
 
-	/// <summary>
-	/// Switches the right click menu state, enabling/disabling the right buttons matching that state
-	/// </summary>
-	/// <param name="newState">The new state detemining what buttons are active/disabled</param>
-	public void SwitchState(ContextState newState)
-	{
-		state = newState;
+		public void SetTargetObject(GameObject newTarget)
+		{
+			targetGameObject = newTarget;
+		}
 
-		//Start by disabling all buttons but disablig their interactibility
-		var buttons = GetComponentsInChildren<Button>();
-		foreach (Button button in buttons)
-			button.interactable = false;
+		/// <summary>
+		/// Start transforming the focus object of our contextmenu
+		/// </summary>
+		public void TransformObject()
+		{
+			//Enable gizmo
 
-		//Now only active the buttons that should be active in this new state
-		foreach (StateButtons stateButtons in buttonsAvailableOnState)
+			//Activate Transform panel in sidemenu
+			ObjectProperties.Instance.OpenPanel(targetGameObject.name);
+			ObjectProperties.Instance.RenderThumbnailFromPosition(CameraModeChanger.Instance.ActiveCamera.transform.position, targetGameObject.transform.position);
+			ObjectProperties.Instance.AddTransformPanel(targetGameObject);
+		}
+
+		/// <summary>
+		/// Disables our right mouse context menu, and resets its state do default
+		/// </summary>
+		void CloseContextMenu()
 		{
-			if (state == stateButtons.state)
+			contextItemsPanel.gameObject.SetActive(false);
+			state = ContextState.DEFAULT;
+		}
+
+		/// <summary>
+		/// Switches the right click menu state, enabling/disabling the right buttons matching that state
+		/// </summary>
+		/// <param name="newState">The new state detemining what buttons are active/disabled</param>
+		public void SwitchState(ContextState newState)
+		{
+			state = newState;
+
+			//Start by disabling all buttons but disablig their interactibility
+			var buttons = GetComponentsInChildren<Button>();
+			foreach (Button button in buttons)
+				button.interactable = false;
+
+			//Now only active the buttons that should be active in this new state
+			foreach (StateButtons stateButtons in buttonsAvailableOnState)
 			{
-				foreach (Button button in stateButtons.activeButtons)
-					button.interactable = true;
-				return;
+				if (state == stateButtons.state)
+				{
+					foreach (Button button in stateButtons.activeButtons)
+						button.interactable = true;
+					return;
+				}
 			}
 		}
-	}
 
-	void Update()
-	{
-		//TODO: replace with centralized input system to avoid conflicts with other scripts polling for the same buttons
-		if (Input.GetMouseButtonUp(1) && !EventSystem.current.IsPointerOverGameObject())
+		void Update()
 		{
-			Appear();
+			//TODO: replace with centralized input system to avoid conflicts with other scripts polling for the same buttons
+			if (Input.GetMouseButtonUp(1) && !EventSystem.current.IsPointerOverGameObject())
```

Comment:
Even een comment voor wanneer je het Input systeem gaat implementeren, voor een button Up event check je in dit geval voor de IAction.Cancelled property. 

Dus er moet dan in Unity's eigen input systeem een action en een action map hiervoor gemaakt worden.